### PR TITLE
update samtools to 1.16.1

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by FriederikeHanssen** [comment]
MIssed a couple, need to also wait for this one: https://github.com/BioContainers/multi-package-containers/pull/2380

**Comment by FriederikeHanssen** [comment]
No idea, what the problem is with bbmap. Ran locally with docker& conda and the version ymls are the same

File: modules/nf-core/star/genomegenerate/main.nf [hunk 1]
```diff
@@ -2,11 +2,10 @@ process STAR_GENOMEGENERATE {
...
     label 'process_high'
 
-    conda (params.enable_conda ? "bioconda::star=2.7.10a bioconda::samtools=1.15.1 conda-forge::gawk=5.1.0" : null)
+    conda (params.enable_conda ? "bioconda::star=2.7.10a bioconda::samtools=1.16.1 conda-forge::gawk=5.1.0" : null)
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/mulled-v2-1fa26d1ce03c295fe2fdcf85831a92fbcbd7e8c2:afaaa4c6f5b308b4b6aa2dd8e99e1466b2a6b0cd-0' :
-        'quay.io/biocontainers/mulled-v2-1fa26d1ce03c295fe2fdcf85831a92fbcbd7e8c2:afaaa4c6f5b308b4b6aa2dd8e99e1466b2a6b0cd-0' }"
-
+        'https://depot.galaxyproject.org/singularity/mulled-v2-1fa26d1ce03c295fe2fdcf85831a92fbcbd7e8c2:1df389393721fc66f3fd8778ad938ac711951107-0' :
+        'quay.io/biocontainers/mulled-v2-1fa26d1ce03c295fe2fdcf85831a92fbcbd7e8c2:1df389393721fc66f3fd8778ad938ac711951107-0' }"
```
Comment (drpatelh) [inline modules/nf-core/star/genomegenerate/main.nf:8]: Need a newline here?

File: modules/nf-core/yara/mapper/main.nf [hunk 2]
```diff
@@ -4,8 +4,8 @@ process YARA_MAPPER {
 
     conda (params.enable_conda ? "bioconda::yara=1.0.2 bioconda::samtools=1.15.1" : null)
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/mulled-v2-f13549097a0d1ca36f9d4f017636fb3609f6c083:d6c969c1e20cc02a9234961c07a24bb0887f05ea-0' :
-        'quay.io/biocontainers/mulled-v2-f13549097a0d1ca36f9d4f017636fb3609f6c083:d6c969c1e20cc02a9234961c07a24bb0887f05ea-0' }"
+        'https://depot.galaxyproject.org/singularity/mulled-v2-f13549097a0d1ca36f9d4f017636fb3609f6c083:de7982183b85634270540ac760c2644f16e0b6d1-0' :
```
Comment (drpatelh) [inline modules/nf-core/yara/mapper/main.nf:7]: Didn't update the Conda version here?

**Review by drpatelh — APPROVED** [review]
Thanks! Couple of minor changes.

**Comment by drpatelh** [comment]
Just make sure the docker / singularity tests pass before merging.

File: modules/nf-core/yara/mapper/main.nf (hunk 2 above)
Comment (FriederikeHanssen) [reply modules/nf-core/yara/mapper/main.nf:7]: thanks, good spot

File: modules/nf-core/star/genomegenerate/main.nf (hunk 1 above)
Comment (FriederikeHanssen) [reply modules/nf-core/star/genomegenerate/main.nf:8]: added

File: modules/nf-core/bwa/sampe/main.nf [hunk 3]
```diff
@@ -2,10 +2,10 @@ process BWA_SAMPE {
     tag "$meta.id"
     label 'process_medium'
 
-    conda (params.enable_conda ? "bioconda::bwa=0.7.17 bioconda::samtools=1.15.1" : null)
+        conda (params.enable_conda ? "bioconda::bwa=0.7.17 bioconda::samtools=1.16.1" : null)
```
Comment (drpatelh) [inline modules/nf-core/bwa/sampe/main.nf:5]: Dodgy indent 👀

File: modules/nf-core/bwa/samse/main.nf [hunk 4]
```diff
@@ -2,10 +2,10 @@ process BWA_SAMSE {
     tag "$meta.id"
     label 'process_medium'
 
-    conda (params.enable_conda ? "bioconda::bwa=0.7.17 bioconda::samtools=1.15.1" : null)
+        conda (params.enable_conda ? "bioconda::bwa=0.7.17 bioconda::samtools=1.16.1" : null)
```
Comment (drpatelh) [inline modules/nf-core/bwa/samse/main.nf:5]: Dodgy indent 👀 👀

**Comment by FriederikeHanssen** [comment]
Can't figure out the issues with bbmap, pushed to an issue to try again later. need these modules here now: #2504

**Pull request merged**